core/commands/name/name.go

```diff
@@ -62,5 +77,159 @@ Resolve the value of a dnslink:
 		"publish": PublishCmd,
 		"resolve": IpnsCmd,
 		"pubsub":  IpnsPubsubCmd,
+		"inspect": IpnsInspectCmd,
+	},
+}
+
+type IpnsInspectValidation struct {
+	Valid  bool    `json:"valid"`
+	Reason string  `json:"reason,omitempty"`
+	Key    peer.ID `json:"key"`
+}
+
+type IpnsInspectEntry struct {
+	Value        string                          `json:"value"`
+	ValidityType *ipns_pb.IpnsEntry_ValidityType `json:"validityType"`
+	Validity     *time.Time                      `json:"validity"`
+	Sequence     uint64                          `json:"sequence"`
+	TTL          *uint64                         `json:"ttl"`
+	PublicKey    string                          `json:"publicKey"`
+	Signature    string                          `json:"signature"`
+	Data         interface{}                     `json:"data"`
```

Comment:
nit: please use exactly the same names as 
https://github.com/ipfs/specs/blob/main/ipns/IPNS.md#record-serialization-format

```suggestion
	Value        string                          `json:"value"`
	ValidityType *ipns_pb.IpnsEntry_ValidityType `json:"validityType"`
	Validity     *time.Time                      `json:"validity"`
	Sequence     uint64                          `json:"sequence"`
	TTL          *uint64                         `json:"ttl"`
	PublicKey    string                          `json:"pubKey"`
	SignatureV1    string                        `json:"signatureV1"`
    SignatureV2    string                        `json:"signatureV2"`
	Data         interface{}                     `json:"data"`
```

this will make it  easier to debug